cql3/restrictions/statement_restrictions.cc

```diff
@@ -608,6 +608,8 @@ void statement_restrictions::calculate_column_defs_for_filtering(data_dictionary
                 }
                 if (!column_uses_indexing(cdef, single_col_restr)) {
                     column_defs_for_filtering.emplace_back(cdef);
+                } else {
+                    _single_column_partition_key_restrictions.erase(it);
```

Comment:
I hope we don't have any code that for some reason needs the old, unmodified, restrictions - be it just for debugging printout or for something more serious. For example, https://github.com/scylladb/scylladb/pull/14450 needs the query to have remained more-or-less unchanged, but isn't damaged by this patch (it happens before this code runs). I just hope we don't have something else that needs it. 
I also I believe/hope our test suite is strong enough in this area so a real regression would be caught.